src/components/SelectableContent.tsx

```diff
@@ -0,0 +1,122 @@
+import FrameworkSwitcher from "@/components/FrameworkSwitcher";
+import Select from "@/components/Select";
+import { getStoredSyncKey, storeSyncKey } from "@/lib/prefs";
+import { getComparator } from "@/lib/utils";
+import type { ForwardedRef, HTMLProps, ReactNode } from "react";
+import { forwardRef, useEffect, useState } from "react";
+
+import styles from "./SelectableContent.module.scss";
+
+type Slot = {
+  key: string;
+  value: ReactNode;
+};
+
+interface Props extends HTMLProps<HTMLDivElement> {
+  syncKey: string;
+  frameworkSwitcher: boolean;
+}
+
+/**
+ * Selectable Content
+ *
+ * Displays content based on user selection of syncKey and framework.
+ *
+ * @param syncKey - The sync key to persist the selection.
+ * @param frameworkSwitcher - Shows the framework switcher.
+ */
+const SelectableContent = forwardRef(
+  (
+    {
+      syncKey,
+      frameworkSwitcher = false,
+      className,
+      children,
+      ...props
+    }: Props,
+    ref: ForwardedRef<HTMLDivElement>,
+  ) => {
+    let cls = "SelectableContent " + styles.SelectableContent;
+    if (className) cls += " " + className;
+
+    const [selectedSlot, setSelectedSlot] = useState<Slot>();
+    const [slots, setSlots] = useState<Slot[]>();
+
+    // Select change callback
+    const onChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+      const val = e.target.value;
+
+      storeSyncKey(syncKey, val);
+
+      if (slots && slots.length > 0) {
+        setSelectedSlot(slots.find((s) => s.key == val));
+      }
+    };
+
+    // Map props to slots
+    useEffect(() => {
+      setSlots(() => {
+        return Object.entries(props)
+          .map((entry) => {
+            return { key: entry[0], value: entry[1] };
+          })
+          .sort(getComparator("desc", "key"));
```

Comment:
Can we make them appear in the order they are in the code? npm should show first, but also we want to show TS (App) for Next.js examples. That's the order they appear in the code, but sorting alphabetically asc or desc won't work for either of those examples.

Reply:
@davidmytton I've improved this to index the slots (see https://github.com/arcjet/arcjet-docs/pull/82/commits/872f8be4dff085d5e8cd442940d843a51d8841eb), also updated everywhere it's used.